### PR TITLE
fix(exportXXX): bind the methods

### LINKED ISSUE #660: exportRules() throws ⚡️ `Cannot read property '_jsonRequest' of undefined`
**Do you want to request a *feature* or report a *bug*?**

bug

**What is the current behavior?**

After loading https://cdn.jsdelivr.net/algoliasearch/3/algoliasearch.min.js (v3.24.9), the JavaScript code below throws `TypeError: Cannot read property '_jsonRequest' of undefined`

```
const fromClient = algoliasearch(fromAppId, fromApiKey);
const fromIndex = fromClient.initIndex(fromIdxName);
fromIndex.exportRules(100, console.log);
```

Tested on both nodejs (back-end) and in jsfiddle (front-end) with the same version.

I relied on the following documentation: https://www.algolia.com/doc/api-reference/api-methods/rules-export/?language=javascript#examples

**What is the expected behavior?**

I expected to have a list query rules printed to my javascript console.

### PR DESCRIPTION
**Pull request by Haroenv**
**Summary**

These two errors were unnoticed, because the integration tests apparently bind everything.

**Result**

fixes #660

No more error about `this.as._jsonRequest` being undefined

### CODE REVIEW

File: test/spec/common/index/exportRules.js [hunk 1]
```diff
@@ -0,0 +1,166 @@
+'use strict';
```
Comment (vvo) [inline test/spec/common/index/exportRules.js]: Why not just a test-case/ https://github.com/algolia/algoliasearch-client-javascript/tree/develop/test/spec/common/client/test-cases?

Comment (Haroenv) [reply test/spec/common/index/exportRules.js]: Because it's not a single API call, it's more like `browseAll`

Comment (vvo) [reply test/spec/common/index/exportRules.js]: got it

**Comment by vvo** [comment]
Can you also add a return here:

https://github.com/algolia/algoliasearch-client-javascript/blob/58a4b2dbe6258769a869e2668d0d66367359504c/src/Index.js#L671-L674

This should be callback(); return;

so that you do not return the promise too (nothing major but still)

**Pull request merged**